integrations/operator/controllers/resources/role_controller_test.go

```diff
@@ -47,21 +48,19 @@ func TestRoleCreation(t *testing.T) {
 	// End of setup, we create the role in Kubernetes
 	k8sCreateDummyRole(ctx, t, setup.K8sClient, setup.Namespace.Name, roleName)
 
+	var tRole types.Role
+	var err error
 	// We wait for the role to be created in Teleport
 	fastEventually(t, func() bool {
```

Comment:
Can you use `require.EventuallyWithT`? 
With it you don't need to expose the roles and err and it still ensures fails with proper messages. It seems a better option to deal with updates instead of creates given that !trace.IsNotFound will  find the role but the version you have there might not be the desired version.

Reply:
Calling `require.Fail()` in `EventuallyWithT` doesn't have the same effect: it marks the specific eventually iteration as failed, not the whole test. Each require.X in the fastEventually function was meant to cover illegal cases. If we ever encounter those cases, we must not retry and fail. Moving them out of eventually is closer to the intended behaviour, I think.